src/Symfony/Component/ObjectMapper/DependencyInjection/MapResourcesPass.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ObjectMapper\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\ObjectMapper\Attribute\Map;
+use Symfony\Component\ObjectMapper\ObjectMapperInterface;
+
+/**
+ * Registers map classes from {@see Map} attribute.
+ *
+ * @author Florent Blaison <florent.blaison@gmail.com>
+ */
+final class MapResourcesPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        if (!class_exists(ObjectMapperInterface::class)) {
+            return;
+        }
+
+        $classes = [];
+        foreach ($container->getDefinitions() as $definition) {
+            if ($definition->hasTag('object_mapper.map')) {
+                $attributes = $definition->getTag('object_mapper.map')[0];
+
+                if (isset($attributes['source'])) {
+                    $classes[$attributes['source']] = $attributes['target'];
+                }
+            }
+        }
+
+        $container->setParameter('object_mapper.map_classes', $classes);
```

Comment:
instead of using a parameter we can just add the argument to the service in this pass. Indeed a parameter should be shared to other bundles but I don't think we want to share this today (unless another bundle has a use case but I doubt it).